common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -497,4 +503,44 @@ public static String hashString(String input) {
         }
     }
 
+    /**
+     * Checks if the input is safe (non-null, non-blank, matches safe character set).
+     *
+     * @param value The input string to validate
+     * @return true if input is safe, false otherwise
+     */
+    public static boolean isSafeText(String value) {
+        if (value == null || value.isBlank()) {
+            return false;
+        }
+        return value.matches(SAFE_INPUT_REGEX);
+    }
+
+    /**
+     * Validates a map of fields to ensure that their values only contain allowed characters.
+     * <p>
+     * Allowed characters are: letters, digits, spaces, underscores (_), hyphens (-), dots (.), and colons (:).
+     * If a value does not comply, a validation error is added.
+     *
+     * @param fields A map where the key is the field name (used for error messages) and the value is the text to validate.
+     * @return An {@link ActionRequestValidationException} containing all validation errors, or {@code null} if all fields are valid.
+     */
+    public static ActionRequestValidationException validateFields(Map<String, String> fields) {
+        ActionRequestValidationException exception = null;
+
+        for (Map.Entry<String, String> entry : fields.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+
+            if (value != null && !isSafeText(value)) {
+                exception = addValidationError(
+                    key + " can only contain letters, digits, spaces, underscores (_), hyphens (-), dots (.), and colons (:)",
```

Comment:
To improve the performance, using StringBuilder msgBuilder = new StringBuilder() for error messages will reduce the memory usage.

Reply:
The compiler (JAVA 9+) uses string concatenation factories (invokedynamic) and optimizes constants

For simple, non-loop cases, a + b + c is often as fast (or faster) than manual StringBuilder.